code/ecdsap256/Hacl.P256.fsti

```diff
@@ -258,16 +311,42 @@ val verify_q:
       )
     )
 
+[@@ Comment "Validate a private key.
+
+Input: scalar: uint8[32].
+  \n Output: bool, where true stands for the scalar to be more than 0 and less than order."]
+val validate_private_key: x: lbuffer uint8 (size 32) -> Stack bool
+  (requires fun h -> live h x)
+  (ensures  fun h0 r h1 -> modifies0 h0 h1 /\
+    (
+      let scalar = nat_from_bytes_be (as_seq h0 x) in 
+      r <==> (scalar > 0 && scalar < prime_p256_order)
+    )
+  )
 
-[@ (Comment " There and further we introduce notions of compressed point and not compressed point. 
-  \n We denote || as byte concatenation. 
-  \n A compressed point is a point representaion as follows: (0x2 + y % 2) || x.
-  \n A not Compressed point is a point representation as follows: 0x4 || x || y.
+[@@ CPrologue "
+/*****************************************/
+/* Point representations and conversions */
+/*****************************************/
+
+/*
+  Elliptic curve points have 2 32-byte coordinates (x, y) and can be represented in 3 ways:
+
+  - \"raw\" form (64 bytes): the concatenation of the 2 coordinates, also known as \"internal\"
+  - \"compressed\" form (33 bytes): the first byte is either \x02 or \x03, followed by x
```

Comment:
I would have put that the first byte is equal to 0x2 + (y % 2); otherwise, it sounds a bit random, wdyt?

Reply:
not sure the details of the representation matter to the clients... I was just aligning myself to what the OCaml API describes. Unless you're implementing P256, do you need to know what `0x02` and `0x03` correspond to for the initial byte? I feel like clients don't need to know about internal representation details.

Reply:
Technical details on the first byte are probably not necessary. But usually this is described as the sign byte or even/odd. Maybe some wording like that would help?

Reply:
What about mentioning https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-186-draft.pdf (D.2 Point Compression section)?

Reply:
To me, as a person who's not an expert on P256, I find the NIST standard quite unhelpful. I feel like you're already an expert of P256 and then you know where the standard is, or you're a client and you don't want to hear about those details. I'll try to rephrase to include sign bit.

Reply:
Ok :)